modules/nextflow/src/main/groovy/nextflow/k8s/client/K8sClient.groovy

```diff
@@ -371,7 +371,15 @@ class K8sClient {
         assert jobName
         final podName = findPodNameForJob(jobName)
         if( podName ) {
-            return podState(podName)
+            try {
+                return podState(podName)
+            } 
+            /* pod might be deleted by control plane just after findPodNameForJob() call
+             * so try fallback to jobState
+             */   
+            catch (NodeTerminationException err) {
+                return jobStateFallback0(jobName)           
```

Comment:
It may be useful to add a debug line to track this to simplify troubleshooting

Reply:
Fixed